### PR TITLE
Common concepts

### PR DESCRIPTION
**Pull request by scott-w**
### Proposed changes
 - Initial index
 - Document that we use `extend` everywhere
 - Initial `getOption` work

Link to the issue: #3063 

### Feedback

Let me know what I've missed - I'll try to get as much as possible done before this gets merged.

### CODE REVIEW

File: docs/basics.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+```
+
+When we instantiate `MyView`, each instance will be given a `.bg-success` class
+with a `myRegion` region created on the `.my-region` element.
+
+### Functions Returning Values
+
+In almost every instance where we can set a value, we can also assign a function
+to figure out the value at runtime. In this case, Marionette will run the
+function on instantiating and use the returned value:
```
Comment (paulfalgout) [inline docs/basics.md]: on instantiation

also might mention something like "with the context of the instance"

File: docs/basics.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+As we can see, almost all of the attributes here can be worked out dynamically.
+In most cases, Marionette will call the function once at instantiation, or first
+render, and preserve the value throughout the lifetime of the View. There are
+some exceptions to this rule - these will be referred to with their respective
+documentation.
+
+There are some exceptions to this general rule that will be documented in the
+relevant places.
+
+## Setting Options
```
Comment (paulfalgout) [inline docs/basics.md]: I definitely think it is worth mentioning and perhaps prioritizing `mergeOptions` along with `getOption` as well.

**Comment by scott-w** [comment]
I've made those changes @paulfalgout - let me know what you think

File: docs/basics.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+You can now create instances of `MyView` with JavaScript's `new` keyword:
+
+```javascript
+var view = new MyView();
+```
+
+### Value Attributes
+
+When we extend classes, we can set a form of class attributes by setting new
```
Comment (rafde) [inline docs/basics.md]: I'm having a difficult time reading this description.
Would this description help?
> When we extend classes, we can provide class attributes with specific values by defining them in the object we pass as the first parameter.

I would also note that the values are shallow copied over to the classes's prototype object. All object type values are passed by reference, so changing any attribute values will result in affect all class instances or extending classes.
I am not sure how to phrase all that in a way that isn't overwhelming, but I'd consider thinking of a way to do so.

File: docs/basics.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+render, and preserve the value throughout the lifetime of the View. There are
+some exceptions to this rule - these will be referred to with their respective
+documentation.
+
+There are some exceptions to this general rule that will be documented in the
+relevant places.
+
+### Function Context
+
+Each function call will receive the class instance as its `this` context - this
```
Comment (rafde) [inline docs/basics.md]: The formatting looks off.
> `this` context - this

or

> `this` context
- this gives you a handy way to ...

or

> `this` context - this gives you a handy way to ...

?

File: docs/basics.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend();
+
+var view = new MyView({
+  foo: 'some text'
+});
+
+view.checkOption();
```
Comment (rafde) [inline docs/basics.md]: I don't know where `view.checkOption();` is coming from.

File: docs/basics.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+This only works for custom options - arguments that belong to the standard
+Backbone/Marionette attributes, such as `model` and `collection`, are not
+accessible via `getOption` and should be accessed as just `view.model` or
+`view.collection`.
+
+### The `mergeOptions` Method
+
+The `mergeOptions` method takes two arguments: an `options` object and `keys` to
+pull from the options object. Any matching `keys` will be merged onto the
+object instance. For example:
```
Comment (rafde) [inline docs/basics.md]: `class instance` for consistency?

**Comment by rafde** [comment]
Sorry for injecting myself in this PR, but I started reading and was hearing the voices of my co-workers who are good at asking questions when they don't understand something. I figure I might as well as commit to annoy you with my suggestions and questions.

File: docs/basics.md (hunk 3 above)
Comment (scott-w) [inline docs/basics.md]: I like it, good call.

File: docs/basics.md (hunk 5 above)
Comment (scott-w) [inline docs/basics.md]: Looks like a typo on my part - I'll look through it again and get it sorted.

**Comment by scott-w** [comment]
Hi @rafde thanks for the comments, it's really helpful to get this feedback before we push it all live. These are exactly the sort of questions we're trying to help answer with this.

**Comment by denar90** [comment]
👍

**Pull request merged**